agent/handlers/task_server_setup_sudo_integ_test.go

```diff
@@ -0,0 +1,427 @@
+			faultType1:    faulttype.BlackHolePortFaultType,
+			faultType2:    faulttype.PacketLossFaultType,
+			body1:         happyBlackHolePortReqBody,
+			body2:         happyNetworkPacketLossReqBody,
+			responseCode1: http.StatusOK,
+			responseCode2: http.StatusOK,
+		},
+	}
+
+	skipForUnsupportedTc(t)
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			// Starting a server which will include the network fault handlers
+			server, port := startServer(t)
+
+			client := &http.Client{}
+
+			// Making request 1
+			bodyBytes, _ := json.Marshal(tc.body1)
+			bodyReader := bytes.NewReader(bodyBytes)
+			req1, err := http.NewRequest(http.MethodPost, getURL(port, fmt.Sprintf(startEndpoint, endpointId, tc.faultType1)), bodyReader)
+			require.NoError(t, err)
+
+			res1, err := client.Do(req1)
+			require.NoError(t, err)
+
+			// Making request 2
+			var resp2Code int
+			// Adding a retry mechanism since there's a rate limit of only one request per 5 seconds for each network fault endpoints
+			// This is in the case of making the same requests type back to back
+			for i := 0; i < requestMaxRetries; i++ {
+				bodyBytes, _ = json.Marshal(tc.body2)
+				bodyReader = bytes.NewReader(bodyBytes)
+				req2, err := http.NewRequest(http.MethodPost, getURL(port, fmt.Sprintf(startEndpoint, endpointId, tc.faultType2)), bodyReader)
+				require.NoError(t, err)
+				res2, err := client.Do(req2)
+				require.NoError(t, err)
+				resp2Code = res2.StatusCode
+
+				if res2.StatusCode != http.StatusTooManyRequests {
+					break
+				}
+				t.Log("Retrying request...")
+				time.Sleep(requestTimeoutInSeconds)
+			}
+
+			// Cleaning up network black hole port fault injections
+			if tc.faultType1 == faulttype.BlackHolePortFaultType {
+				cleanupBlackHolePortFault(t, port, tc.body1)
+			}
+			if tc.faultType2 == faulttype.BlackHolePortFaultType {
+				// Adding a retry mechanism to clean up the network black hole port fault injection from the second request
+				for i := 0; i < requestMaxRetries; i++ {
+					code := cleanupBlackHolePortFault(t, port, tc.body2)
+					if code != http.StatusTooManyRequests {
+						break
+					}
+					time.Sleep(requestTimeoutInSeconds)
+				}
+			}
+			// Cleaning up network latency and network packet loss fault injections
+			cleanupLatencyAndPacketLossFaults(t, port)
+
+			shutdownServer(t, server)
+			assert.Equal(t, tc.responseCode1, res1.StatusCode)
+			assert.Equal(t, tc.responseCode2, resp2Code)
+		})
+	}
+}
+
+// skipForUnsupportedTc will test whether or not the tc utility installed can use the required flag option(s) [e.g. -j] which is used to start/stop/check status of certain
+// network faults such as latency and packet loss.
+func skipForUnsupportedTc(t *testing.T) {
+	cmdCtx, cancel := context.WithTimeout(context.Background(), requestTimeoutInSeconds)
+	defer cancel()
+
+	_, err := exec.CommandContext(cmdCtx, "bash", "-c", "tc -j q show").CombinedOutput()
```

Comment:
nit: Why do we need `bash -c`? Why does `exec.CommandContext(cmtCtx, "tc", "-j", "q", "show")` not work?

Reply:
sure, I will remove the `bash -c`.